arteria/models/runfolder_utils.py

```diff
@@ -0,0 +1,177 @@
+import os
+import re
+import time
+import logging
+import xmltodict
+
+from pathlib import Path
+from arteria.models.state import State
+
+log = logging.getLogger(__name__)
+
+
+def list_runfolders(monitored_directories, filter_key=lambda r: True):
+    """
+    Returns list of Runfolders in the monitored_directories
+    according to the state filter provided (filter_key), or all
+    runfolders when no state filter is given.
+    """
+    runfolders = []
+    for monitored_directory in monitored_directories:
+        for subdir in os.listdir(monitored_directory):
+            try:
+                if filter_key(Runfolder(os.path.join(monitored_directory, subdir))):
+                    runfolders.append(Runfolder(os.path.join(monitored_directory, subdir)))
```

Comment:
```suggestion
                if filter_key(runfolder := Runfolder(os.path.join(monitored_directory, subdir))):
                    runfolders.append(runfolder)
```
You can use the [walrus operator](https://realpython.com/python-walrus-operator/) to avoid instantiating the same runfolder twice here :)